### PR TITLE
Add JS parser initialisation function and config struct

### PR DESCRIPTION
**Pull request by maxfisher-g**
The initialisation involves writing the parser to a custom directory. This way no dependencies have to be installed beforehand (apart from NPM)

### CODE REVIEW

File: internal/staticanalysis/parsing/js/js_parsing_test.go [hunk 1]
```diff
@@ -299,11 +298,18 @@ var index = 0,
 
 func TestParseJS(t *testing.T) {
 	const printAllJson = false
+	const jsParserDir = "/tmp/package-analysis-js-parsing-test-parser"
```
Comment (calebbrown) [inline internal/staticanalysis/parsing/js/js_parsing_test.go:301]: You probably want to use `t.TempDir()` here - it is unique per test invocation, and is automatically cleaned up by Go.

File: internal/utils/write_file.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+In more detail:
+
+ 1. If a non-regular file exists at path, an error is returned.
+ 2. If a regular file exists at path, it is truncated and marked as executable if applicable.
+ 3. If no file exists at path, a new one is created with executable permissions if applicable
+
+In cases 2 and 3 above, the contents of scriptSource are then written to the file.
+*/
+func WriteFile(path string, contents []byte, executable bool) (err error) {
```
Comment (oliverchang) [inline internal/utils/write_file.go:22]: Can we just use https://pkg.go.dev/io/ioutil#WriteFile instead?

**Review by calebbrown — DISMISSED** [review]

File: internal/utils/write_file.go (hunk 2 above)
Comment (calebbrown) [reply internal/utils/write_file.go:22]: Or `os.WriteFile` with a small wrapper to fix perms.

File: internal/utils/write_file.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	if err != nil && !errors.Is(err, fs.ErrNotExist) {
+		return fmt.Errorf("could not stat %s: %v", path, err)
+	}
+
+	var fileMode os.FileMode
+	if executable {
+		fileMode = 0700
+	} else {
+		fileMode = 0600
+	}
```
Comment (calebbrown) [inline internal/utils/write_file.go:46]: I suggest using fileMode `0777` or `0666` and let umask sort out the rest

Comment (maxfisher-g) [reply internal/utils/write_file.go:46]: Will do!

File: internal/utils/write_file.go (hunk 2 above)
Comment (maxfisher-g) [reply internal/utils/write_file.go:22]: Originally I this function was only for executable script files, and I had a check that if the file was nonempty and nonexecutable, I'd return an error, as well as the one for non-regular files. I can remove the check for a non-regular file and unconditionally truncate if that's an acceptable simplification.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**